File: python/추석트래픽.py


def solution(lines):
    answer = 0
    s_time = []
    e_time = []

    for t in lines:
        time = t.split(" ")
        s_time.append(get_start(time[1], time[2]))
        e_time.append(to_time(time[1]))
    for i in range(len(lines)):
        cnt = 0
        cur = e_time[i]
        for j in range(i, len(lines)):
            if cur-s_time[j] < 1000:
                cnt += 1
        answer = max(answer, cnt)
    return answer


def to_time(time):
    hour = int(time[:2]) * 3600
    minute = int(time[3:5]) * 60
    second = int(time[6:8])
    millisecond = int(time[9:])
    print(millisecond)
    print((hour + minute + second) * 1000 + millisecond)


def get_start(time, dur):
    m_time = dur[:-1]
    int_sec = int(float(m_time) * 1000)
    return to_time(time) - int_sec+1

File: cpp/newWave/추석 트래픽.cpp
#include<bits/stdc++.h>
#define MIN 60
#define SEC 60
#define MILLISEC 1000
using namespace std;

int solution(vector<string> lines) {
    int ans = 0;
    vector<int> startAt, finishAt;
    for(auto line : lines){
        int hour , minute, second , millisec, time;
        hour = stoi(line.substr(11,2))* MIN * SEC *MILLISEC;
        minute = stoi(line.substr(14,2)) * SEC*MILLISEC;
        second = stoi(line.substr(17,2))*MILLISEC;
        millisec = stoi(line.substr(20, 3));
        
        
        line.pop_back();
        time = stof(line.substr(24))* MILLISEC;
        
          
        startAt.push_back(hour + minute + second + millisec - time + 1);
        finishAt.push_back(hour + minute + second + millisec);
    }
    
    
    for(int i=0; i<finishAt.size(); i++){
        finishAt[i] +=1000;
        int cnt=0;
        
        
        for(int j=i; j<startAt.size(); j++) if(startAt[j] < finishAt[i]) cnt++;
        ans= max(ans,cnt);
    }
    
    
    return ans;
}
